src/arr/trove/ast.arr

```diff
@@ -945,6 +1015,19 @@ data Expr:
   | s-str(l :: Loc, s :: String) with:
     method label(self): "s-str" end,
     method tosource(self): PP.str(torepr(self.s)) end
+    # Base should be a name that refers to a module import, like import lists as L
+    # path is generalized to list because eventually this should support
+    # Lib.Mod.x kinds of references
+    #
+    # NOTE: The following invariants should hold for any s-module-dot 'smd':
+    #   is-s-mref-by-name(smd.base) => smd.path.length() >= 1
+    #   is-s-mref-by-uri(smd.base)  => smd.path.length() == 1
```

Comment:
I think there's a different invariant here. Pre-scope-resolution, this could work either way.

Post-scope-resolution, everything should have a path of length 1, and a base that is a `mref-by-uri`.

This seems complicated! Why isn't this just an inductively collapsing case in resolve-scope that _always_ uses a `mref-by-uri`?